### PR TITLE
Add basic share button functionality as a reusable UIViewController that manages a UIButton

### PR DESCRIPTION
**Pull request by JoeyCarson**
Add basic share button functionality as a reusable UIViewController that manages a UIButton.  At the moment the only thing that is shared is the podcast URL but it can be extended by the caller to include more information if desired.

### CODE REVIEW

**Comment by themisterholliday** [comment]
Cool! I'll take a look at this in depth in a bit.

File: SEDaily-IOS/ShareButtonViewController.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        button.setTitle("Share", for: .normal)
+        button.setBackgroundColor(color: Stylesheet.Colors.shareButtonDefault, forState: .normal)
+        button.cornerRadius = UIView.getValueScaledByScreenHeightFor(baseValue: 4)
+        
+        view = button
+    }
+    
+    @objc func shareButtonPressed() {
+        SwiftyBeaver.info("Share UI Button Pressed")
+        let shareUI = SharePodcastViewController(activityItems: [shareObj!], applicationActivities: nil)
```
Comment (justinctlam) [inline SEDaily-IOS/ShareButtonViewController.swift]: Should consider checking shareObj before force unwrapping.

**Comment by JoeyCarson** [comment]
Another detail folks.  My macbook is newer than my iPhone and I don't have a USB-C cable, so all of my testing was done on the simulator.  This means that I couldn't install an app from the store, e.g. facebook or Twitter.  My testing was primarily to copy the podcast URL.  If deployed on an iPhone with facebook, twitter, etc installed, those services should accept the URL, though I haven't verified it.  

So if you can, sync down these changes and build and deploy to a device with the aforementioned apps installed and verify that the URL is shared to those services.

File: SEDaily-IOS/HeaderView.swift [hunk 2]
```diff
@@ -81,38 +84,30 @@ class HeaderView: UIView {
...
-            make.centerY.equalToSuperview()
-            make.right.equalToSuperview().inset(UIView.getValueScaledByScreenWidthFor(baseValue: 15))
-            make.width.equalTo(UIView.getValueScaledByScreenWidthFor(baseValue: 84))
-            make.height.equalTo(UIView.getValueScaledByScreenHeightFor(baseValue: 42))
-        }
-
-        playView.addSubview(voteView)
+        
+        // vote view.
+        //voteView.backgroundColor = UIColor.orange
```
Comment (TheHollidayInn) [inline SEDaily-IOS/HeaderView.swift:100]: Do you need this line of code? I don't see it in the old version.

Comment (JoeyCarson) [reply SEDaily-IOS/HeaderView.swift:100]: Certainly not.  Just had it there for showing UI boundaries easier. Will remove it.

Comment (TheHollidayInn) [reply SEDaily-IOS/HeaderView.swift:100]: Oh I see the code was there from before. You can ignore my comment :P

**Comment by TheHollidayInn** [comment]
Tested locally and it is working great!

Would you be able to add the `featuredImageURL` property for sharing the image as well?

Also, I think I'd prefer if we place the share button in the menu like in the right screen of this picture (and use the default share icon)
https://i.amz.mshcdn.com/Ukg_i3vyZB2DFuJToFqcjqUt0WM=/fit-in/1200x9600/http%3A%2F%2Fmashable.com%2Fwp-content%2Fuploads%2F2015%2F04%2FApple-Watch-App-Store-1-640x565.jpg

**Comment by JoeyCarson** [comment]
Hi, sure thing.  I will try to get to this over the weekend.  Sorry for the delay.  Busy with work, school, and other stuff.

**Comment by TheHollidayInn** [comment]
No rush! Thanks for working on it :D

**Pull request closed without merging**